javascript/signin.js

```diff
@@ -0,0 +1,122 @@
+document.addEventListener('DOMContentLoaded', function () {
+  const form = document.querySelector('.sign-box');
+  const emailInput = document.querySelector('#signin-email');
+  const passwordInput = document.querySelector('#signin-password');
+  const errorTextEmail = document.createElement('div');
+  const errorTextPassword = document.createElement('div');
+  const passwordCover = document.querySelector('.eyecon');
+  let isPasswordCovered = false;
+
+  //////////////// 함수 정의 ////////////////////
+
+  // 이메일 확인 함수
+  function emailCheck() {
+    const email = emailInput.value.trim();
+    clearErrorStyle(emailInput);
+
+    // 기존 에러 메세지 삭제
+    if (errorTextEmail.parentNode === emailInput.parentNode) {
+      emailInput.parentNode.removeChild(errorTextEmail);
+    }
+    // 이메일 미입력 에러
+    if (email == '') {
+      showError(errorTextEmail, '이메일을 입력해 주세요.', emailInput);
+      return false;
+    }
+    // 이메일 형식 확인
+    const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+    if (!emailRegex.test(email)) {
+      showError(errorTextEmail, '올바른 이메일 주소가 아닙니다.', emailInput);
+      return false;
+    }
+    return true;
+  }
+
+  // 비번 확인 함수
+  function passwordCheck() {
+    const password = passwordInput.value.trim();
+    clearErrorStyle(passwordInput);
+    // 기존 에러 메세지 삭제
+    if (errorTextPassword.parentNode === passwordInput.parentNode) {
+      passwordInput.parentNode.removeChild(errorTextPassword);
+    }
+    // 비밀번호 미입력 에러
+    if (password == '') {
+      showError(errorTextPassword, '비밀번호를 입력해 주세요.', passwordInput);
+      return false;
+    }
+    return true;
+  }
+
+  // 에러 스타일 빼기
+  function clearErrorStyle(inputElement) {
+    const parentContainer = inputElement.parentNode;
+    const errorElement = parentContainer.querySelector('.error-message');
+
+    parentContainer.classList.remove('error');
+    if (errorElement == true) {

```

Comment:
자바스크립트에서 `==` 은 되도록 사용해주지 않은 편이 좋습니다! 자세한 내용은 아래 포스팅 첨부해보시면 좋을거예요!

```suggestion
    if (errorElement === true) {

```
https://velog.io/@najiexx/JavaScript-%EC%99%80-%EC%9D%98-%EC%B0%A8%EC%9D%B4%EC%A0%90%EB%8F%99%EB%93%B1%EC%97%B0%EC%82%B0%EC%9E%90%EC%99%80-%EC%9D%BC%EC%B9%98%EC%97%B0%EC%82%B0%EC%9E%90

Reply:
앗 무의식적으로 쓰다보니 오타가 생겼네요..;; 덕분에 ===에 대해서 정확히 파악하게 됐어요😋 수정했습니다! c97a1e1b40c7e37083634ef31c99d77908939a79